### PR TITLE
Fix enumeration & null queries

### PR DESCRIPTION
**Pull request by jasonaowen**
I introduced a bug in 2c050e253528bea510035933348855dc48764991, but it was hidden behind other Hibernate 5 migration problems. When I examined the broken code more closely, I found a second bug.

The first is a piece I missed from converting the `applicantType` field from an `int` to the `enum` we had, and allowing Hibernate to serialize that as a string. The fix is to use a dynamic query with a bound parameter instead of hardcoded queries. Also fix a suppressed warning by using [a newer query API](https://docs.oracle.com/javaee/7/api/javax/persistence/EntityManager.html#createQuery-javax.persistence.criteria.CriteriaQuery-) that returns a typed result instead of the older, untyped result.

The second bug is related to `null` parameters, which was explicitly passed in somewhere. In Java, `switch`ing on an `enum` instance that is `null` causes a `NullPointerException`. Split the intended purpose of the (incorrectly unreachable) code into a new method and call that, instead.

Tested by creating a new provider user and starting the enrollment process. I'm not sure how to reach the code that was invoking the second bug; it might be hidden behind another page broken by the Hibernate migration.

### CODE REVIEW

File: psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/LookupServiceBean.java [hunk 1]
```diff
@@ -70,6 +70,13 @@
     public LookupServiceBean() {
     }
```
Comment (cecilia-donnelly) [inline psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/LookupServiceBean.java]: Can you add a code comment here saying what you tell us in the log message?  It's pretty clear, but I think it's a good practice to document each function.

File: psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/LookupServiceBean.java [hunk 2]
```diff
@@ -70,6 +70,13 @@
...
 
+    public List<ProviderType> getProviderTypes() {
+        return em.createQuery(
+                "FROM ProviderType",
+                ProviderType.class
```
Comment (cecilia-donnelly) [inline psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/LookupServiceBean.java]: The `ProviderType.class` is the resultClass as described [here](http://docs.oracle.com/javaee/6/api/javax/persistence/EntityManager.html#createQuery%28java.lang.String,%20java.lang.Class%29), right?  We get multiple results, but they're all `ProviderType`s.  Then `getResultList` turns this into a useable list of objects?

Comment (jasonaowen) [inline psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/LookupServiceBean.java]: Right!

File: psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/LookupServiceBean.java (hunk 1 above)
Comment (jasonaowen) [inline psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/LookupServiceBean.java]: I'm not sure I follow - what would you like to see in the comment? Would renaming the method to something like `getAllProviderTypes` help? (I went back and forth on that name...)

Comment (cecilia-donnelly) [inline psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/LookupServiceBean.java]: That would help!  We generally try to give a doc string for each function/method we write that includes any parameters and what the return value is.  (Actually, @kfogel, should that go in CONTRIB?)  Something like:

/*
     * Retrieves a list of all provider types.
     * 
     * @return the provider types
*/

**Comment by cecilia-donnelly** [comment]
I talked to @jasonaowen and we decided that changing the method name to `getAllProviderTypes` will make the code clear enough.

More generally, I want more comments than Jason does (he find they clutter the code, while I think adding comments as a rule helps disambiguate when they'd be helpful / helps a person skimming the codebase), so we'll probably have more discussions about it.

**Comment by jasonaowen** [comment]
Method renamed!

A lot of my thinking around comments has been informed by chapter 4 of [Clean Code](https://smile.amazon.com/dp/0132350882/Clean-Code-Handbook-Software-Craftsmanship/). Neither I nor the author are completely against comments; the guideline is that a comment should add something constructive that can't reasonably and readably be expressed in the code.

@cecilia-donnelly thank you for talking about this with me. I look forward to more discussions!

**Pull request merged**